cibuildwheel/platforms/ios.py

```diff
@@ -636,43 +648,39 @@ def build(options: Options, tmp_path: Path) -> None:
 
                                     Test command: {build_options.test_command!r}
                                 """)
+                                log.warning(msg)
+                                final_command = ["pytest", *rest]
+                            case _:
+                                msg = unwrap_preserving_paragraphs(
+                                    f"""
+                                        iOS tests configured with a test command which doesn't start
+                                        with 'python -m'. iOS tests must execute python modules - other
+                                        entrypoints are not supported.
+
+                                        Test command: {build_options.test_command!r}
+                                    """
+                                )
+                                raise errors.FatalError(msg)
+
+                        try:
+                            call(
+                                "python",
+                                testbed_path,
+                                "run",
+                                *(["--verbose"] if build_options.build_verbosity > 0 else []),
+                                "--",
+                                *final_command,
+                                env=test_env,
                             )
-                        else:
-                            msg = unwrap_preserving_paragraphs(
-                                f"""
-                                    iOS tests configured with a test command which doesn't start
-                                    with 'python -m'. iOS tests must execute python modules - other
-                                    entrypoints are not supported.
-
-                                    Test command: {build_options.test_command!r}
-                                """
-                            )
-                            raise errors.FatalError(msg)
-                    else:
-                        # the testbed run command actually doesn't want the
-                        # python -m prefix - it's implicit, so we remove it
-                        # here.
-                        test_command_parts = test_command_parts[2:]
-
-                    try:
-                        call(
-                            "python",
-                            testbed_path,
-                            "run",
-                            *(["--verbose"] if build_options.build_verbosity > 0 else []),
-                            "--",
-                            *test_command_parts,
-                            env=test_env,
-                        )
-                        failed = False
-                    except subprocess.CalledProcessError:
-                        failed = True
+                            failed = False
+                        except subprocess.CalledProcessError:
+                            failed = True
```

Comment:
I guess we should break out of the loop here. Or the CalledProcessError exception should be caught outside of the loop - that would negate the need for the `failed` flag I think.

Reply:
A test that checks the failure short-circuits the rest of the command would be good, too.

Reply:
Yeah, I get where you're coming from, but I don't think it's cibuildwheel's place to redefine what `&&` means. As idioms go, it's pretty widespread. 

(we can't really change the delimiter either, because we've always accepted string-style options with `&&` as the delimiter.)